nomad/structs/services.go

```diff
@@ -632,6 +694,11 @@ func (s *Service) Hash(allocID, taskName string, canary bool) string {
 	hashString(h, s.OnUpdate)
 	hashString(h, s.Namespace)
 
+	// Don't hash the provider parameter, so we don't cause churn of all
+	// registered services when upgrading Nomad versions. The provider is not
+	// used at the level the hash is and therefore is not needed to tell
+	// whether the service has changed.
```

Comment:
Let's include `Provider is not hashed` in the func's comment as well since we mention checks there. Great comment though. :+1: